### DISCUSSION TITLE
win10 生成 dll 尺寸太大，怎么优化

### DISCUSSION THREAD

**Discussion by penghongquan** [xmake-io/xmake#4844] [Q&A] [answered]
我使用 xmake create -l c -t shared my_dll 生成例子后 xmake 编译，发现生成的 dll 比较大认为有异常，请问怎么优化呢？

**Comment by waruqi** [comment]
多大，我看也只有 107k 左右，很正常么。。vs 生成的也要 108k

**Reply by penghongquan** [reply]
因为我使用 cmake 编译的话，才 9k 所以我认为不正常

**Reply by waruqi** [reply, marked as answer]
xmake 默认不添加 vs runtimes flags，msvc 默认用的是 MT，cmake 默认用的是 MD ，没可比性。。

你可以切到 `set_runtimes("MD")` 编译

**Reply by penghongquan** [reply]
谢谢，我对这块不熟悉，现在切到 set_runtimes("MD") 尺寸一致了，我去查查这俩的区别

**Comment by shuax** [comment]
安利一下 https://github.com/Chuyu-Team/VC-LTL5 可以降低程序大小
